src/Version/Version.cpp

```diff
@@ -89,8 +89,16 @@ std::string getOnnxMlirFullVersion(bool toIncludeLLVM) {
 #endif
   std::string repo = getOnnxMlirFullRepositoryVersion(toIncludeLLVM);
   if (!repo.empty()) {
-    os << " " << repo;
+    os << " " << repo << '\n';
   }
+#ifdef ONNX_MLIR_PRODUCT_VERSION
+  os << "onnx-mlir product version " ONNX_MLIR_PRODUCT_VERSION;
```

Comment:
Instead of hardcoding "onnx-mlir product version" this should use the `ONNX_MLIR_VENDOR` (which works out with the earlier comment of moving it up in the order). So maybe something like:
```
#if defined(ONNX_MLIR_VENDOR) || defined(ONNX_MLIR_PRODUCT_VERSION)
os << ONNX_MLIR_VENDOR << " " << ONNX_MLIR_PRODUCT_VERSION << ", "
#endif
```

(opinion) I used an OR because I think if someone specifies one, they'll likely want both in the end. I figured OR will make it easier because if you initially/accidentally only set one, at least they'd see the one they did set. Otherwise we could do an AND but then it wouldn't be obvious (without looking at the code) to understand why their change isn't showing up if they did set just one initially.

I wouldn't be against a more complicated statement that better handles one or the other alone but my gut thought is that might be overkill. I'm open to other thoughts, that's just my initial feelings.

Reply:
I see your thought process... I would lean towards using `&&` because we would want the vendor name and the product version to be displayed and not one or the other. It some what forces the user to have to specify both flags because I can see someone indicating one and not the other even by accident.

Reply:
> It some what forces the user to have to specify both flags because I can see someone indicating one and not the other even by accident.

This is where I was waffling myself. On the one hand, it'd make sure both are specified. On the other hand, if someone only specifies one, it won't be obvious why it has no effect unless they dig into the code so it might take a while to debug what is a 5 second fix on their end.

If we go with `&&` is there a way to make it a build error if one is specified and not the other? Then we could give a useful error.

Reply:
Hmm I might have an alternative idea but you can let me know your thoughts once I push the code updates :)